### PR TITLE
1/init

### LINKED ISSUE #1: Add file providers

### PR DESCRIPTION
**Pull request by Julien-Torrent**

### CODE REVIEW

File: LICENSE copy [hunk 1]
```diff
@@ -0,0 +1,661 @@
+                    GNU AFFERO GENERAL PUBLIC LICENSE
+                       Version 3, 19 November 2007
+
```
Comment (pyphilia) [inline LICENSE copy]: LICENSE copy? We should remove this file I think.

File: package.json [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+  "license": "ISC",
+  "bugs": {
+    "url": "https://github.com/graasp/graasp-plugin-file/issues"
+  },
+  "homepage": "https://github.com/graasp/graasp-plugin-file#readme",
+  "dependencies": {
+    "aws-sdk": "^2.793.0",
+    "content-disposition": "0.5.3",
+    "fastify-multipart": "5.1.0",
+    "graasp-file-upload-limiter": "github:graasp/graasp-file-upload-limiter",
```
Comment (pyphilia) [inline package.json]: We don't need this anymore?

File: package.json [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    "http-status-codes": "2.1.4",
+    "sharp": "0.29.1"
```
Comment (pyphilia) [inline package.json]: This neither.

File: src/fileServices/interface/fileService.ts [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+
+  // get file buffer, used for generating thumbnails
+  getFileBuffer(args: { filepath: string }): Promise<Buffer>;
+
+  downloadFile(args: {
+    reply: FastifyReply;
+    filepath: string;
+    itemId: string;
+    mimetype: string;
+  }): Promise<ReadStream> | Promise<void>;
```
Comment (pyphilia) [inline src/fileServices/interface/fileService.ts]: Let's alphabetically reorder this?

File: src/task-manager.ts [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+  createCopyFileTask(member: Actor, data: CopyInputType): Task<Actor, unknown> {
+    // new filename is new item id
+    return new CopyFileTask(member, this.fileService, data);
+  }
+
+  createGetFileBufferTask(
+    member: Actor,
+    data: GetFileBufferInputType,
+  ): Task<Actor, unknown> {
+    // new filename is new item id
```
Comment (pyphilia) [inline src/task-manager.ts]: Is this still valid?

File: src/task-manager.ts [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+
+  createDeleteFolderTask(
+    member: Actor,
+    data: DeleteFolderInputType,
+  ): Task<Actor, unknown> {
+    return new DeleteFolderTask(member, this.fileService, data);
+  }
+
+  createCopyFileTask(member: Actor, data: CopyInputType): Task<Actor, unknown> {
+    // new filename is new item id
```
Comment (pyphilia) [inline src/task-manager.ts]: Is this still valid?

File: src/utils/constants.ts [hunk 7]
```diff
@@ -0,0 +1,6 @@
+export const SMALL = 'small';
+export const MEDIUM = 'medium';
+export const LARGE = 'large';
+export const ORIGINAL = 'original';
+
+export const THUMBNAIL_SIZES = [SMALL, MEDIUM, LARGE, ORIGINAL];
```
Comment (pyphilia) [inline src/utils/constants.ts:6]: I don't think we need this here.

File: src/utils/errors.ts [hunk 8]
```diff
@@ -0,0 +1,123 @@
...
+      data,
+    );
+  }
+}
+
+export class LocalFileNotFound extends GraaspBaseError {
+  constructor(data?: unknown) {
+    super(
+      {
+        code: 'GPFERR005',
```
Comment (pyphilia) [inline src/utils/errors.ts]: Double `GPFERR005`

File: src/utils/errors.ts [hunk 9]
```diff
@@ -0,0 +1,123 @@
...
+      data,
+    );
+  }
+}
+
+export class GetFileBufferInvalidParameterError extends GraaspBaseError {
+  constructor(data?: unknown) {
+    super(
+      {
+        code: 'GPFERR001',
```
Comment (pyphilia) [inline src/utils/errors.ts]: Update code.

**Review by pyphilia — CHANGES_REQUESTED** [review]

File: src/utils/constants.ts (hunk 7 above)
Comment (Julien-Torrent) [reply src/utils/constants.ts:6]: This is needed because the thumbnails can't extend the download schema

File: LICENSE [hunk 10]
```diff
@@ -1,661 +0,0 @@
-                    GNU AFFERO GENERAL PUBLIC LICENSE
-                       Version 3, 19 November 2007
-
- Copyright (C) 2007 Free Software Foundation, Inc. <https://fsf.org/>
- Everyone is permitted to copy and distribute verbatim copies
- of this license document, but changing it is not allowed.
```
Comment (pyphilia) [inline LICENSE]: I think this file should remain in the repo.

**Review by pyphilia — APPROVED** [review]

**Pull request merged**